### ISSUE TITLE
DB/Gossip: Apothecary Dithers and Shadow Priestess Vandis - Mission Accomplished!

### ISSUE THREAD

**Issue by ZenoX92** [TrinityCore/TrinityCore#20523]
**Description:**

Apothecary Dithers and Shadow Priestess Vandis's gossip texts should change after quest "Mission Accomplished! (Horde)" is finished.

**Branch(es):**

3.3.5

**TC rev. hash/commit:** 

TrinityCore rev. 9fa6885 2017-08-08 11:36:46 -0300 (3.3.5 branch) (Win64, Release, Static) (worldserver-daemon)

**TDB version:**  335.63

**Operating system:** Windows 7

**Suggested Fix:**
```sql
-- Apothecary Dithers & Shadow Priestess Vandis --> Fix gossip texts
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`= 14 AND `SourceGroup` IN (3183,3223);
INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`NegativeCondition`,`ErrorType`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
(14,3183,4040,0,0,8,0,5237,0,0,0,0,0,'',"Show Gossip Menu Text 4040 if Quest 'Mission Accomplished!' is rewarded"),
(14,3223,3979,0,0,8,0,5237,0,0,0,0,0,'',"Show Gossip Menu Text 3979 if Quest 'Mission Accomplished!' is rewarded");

**Comment by tkrokli** [comment]
Suggestion for how this SQL can be simplified (and quite easy too) :
```sql
-- gossip conditions for Apothecary Dithers & Shadow Priestess Vandis
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`= 14 AND `SourceGroup` IN (3183,3223); -- no existing condition for this group yet
INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`NegativeCondition`,`ErrorType`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
(14,3183,4040,0,0,8,0,5237,0,0,0,0,0,'',"Show Gossip Menu Text 4040 if Quest 'Mission Accomplished!' is rewarded"),
(14,3223,3979,0,0,8,0,5237,0,0,0,0,0,'',"Show Gossip Menu Text 3979 if Quest 'Mission Accomplished!' is rewarded");
```
(feel free to comment and tell me if I have overlooked any reason for not doing it like this)

**Comment by tkrokli** [comment]
Funny, this gives me flashbacks to when I updated the quest script for Dithers & Arbington ;-)

**Comment by ZenoX92** [comment]
@Wyreth posted a better fix for this here #20540.
